### ISSUE TITLE
!include not inlined in markdown preview

### ISSUE THREAD

**Issue by mada-github** [qjebbs/vscode-plantuml#490]
The plantUML server responds with 
```
[From string (line2)]

@startuml
!include config_sequence.puml
cannot include config_sequence.puml
```
i.e. the include file is not inlined

- VScode workspace folder includes the folder to example.md
- no folder settings in plantuml extension
- example.md and config_sequence.puml are stored in the same folder
- works with markdown preview enhanced (just to verify that files exist, etc...)

the behavior can be reproduced with a minimal md file, e.g.  
example.md
```
    ```plantuml
    @startuml
    !include config_sequence.puml
    @enduml
    ```
```

Is there anything in this setting that forbids include inlining?

**Comment by mada-github** [comment]
This seems to be related to previous redesigns of include handling, i.e. #260.
According to the readme, I would have expected it to work given the new search order: 
1) The folder of the rendered file
2) The includepaths as configured in settings.json
Can somebody confirm whether a relative link (relative to the .md file, that is) is supposed to work in markdown preview?

**Comment by qjebbs** [comment]
No, the rendering codes has no way to get the parent document where the diagram is in, so relative links won't work.

**Comment by wrzlbrzl** [comment]
that is understood from a server point of view.

Are you saying that the [preprocessing](https://github.com/qjebbs/vscode-plantuml/tree/master/src/plantuml/diagram/include.ts) is not used for markdown preview at all?

**Comment by ipcoder** [comment]
I am not able to run it as Local. And not with !include with server on Linux

**Comment by arwedus** [comment]
@qjebbs: I spent half an hour analyzing the code in the [preprocessing](https://github.com/qjebbs/vscode-plantuml/tree/master/src/plantuml/diagram/include.ts) and I don't understand why this could not work.

Here are some ideas:

1. If the parent document is unknown, just try all combinations of {includePaths, diagramsRoot, workspaceRoot} + include_string - the first that is an existing file is used.
2. Alternatively (or, if step 1 fails), search for the relative path of the `!include` in the diagramsRoot, and return the first hit.

Solution \#2 would also make it unnecessary to add includePaths 🙂
